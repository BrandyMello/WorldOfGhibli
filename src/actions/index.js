export const storeFilms = (films) => ({
  type: 'STORE_FILMS',
  films
});

export const storeCharacters = (characters) => ({
  type: 'STORE_CHARACTERS',
  characters
});

export const storeCharNames = (characterNames) => ({
  type: 'STORE_CHAR_NAMES',
  characterNames
});

export const setCurrentPlayer = (playerName) => ({
  type: 'SET_CURRENT_PLAYER',
  playerName
});

export const setWins = (numOfWins) => ({
  type: 'SET_WINS',
  numOfWins
})