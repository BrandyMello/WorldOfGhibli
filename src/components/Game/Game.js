import React from 'react';
import './Game.scss';

const Game = ({ name, gender, age, eyeColor, hairColor, characterNames, player }) => {
  // console.log({characterNames})
  return (
    <main className="game-background-img whole-game">
      <article className="ea-game">
        <h2>{name}</h2>
      {/* <img src={photo} className='movie-image'></img> */}
        <p>Gender: {gender}</p>
        <p>Age: {age}</p>
        <p>Eye Color: {eyeColor}</p>
        <p>Hair Color: {hairColor}</p>
      </article>
    </main>
  )
}

export default Game
