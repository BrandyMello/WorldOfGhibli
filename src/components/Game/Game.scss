.game-background-img {
  // opacity: .4;

  background: url("https://cdn.suwalls.com/wallpapers/anime/studio-ghibli-characters-36913-3840x2160.jpg") no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  z-index: 0;
}

.ea-game {
  height: 220px;
  width: 220px;
  background-color: black;
  border-radius: 50%;
  margin-left: 10%;
  margin-top: 45px;
}

p {
  padding-left: 35px;
}

.background {
  margin-top: 57px;
}

h2 {
  background-color: #6577C3;
  text-align: center;
  margin: 0;
  padding-top: 5px;
}