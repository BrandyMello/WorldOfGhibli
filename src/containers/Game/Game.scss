.game-background-img {
  // background: url("https://cdn.suwalls.com/wallpapers/anime/studio-ghibli-characters-36913-3840x2160.jpg") no-repeat center center fixed;
  // -webkit-background-size: cover;
  // -moz-background-size: cover;
  // -o-background-size: cover;
  // background-size: cover;
  z-index: 0;
}

.ea-game {
  background-color: black;
  border-radius: 50%;
  height: 260px;
  margin-left: 10%;
  margin-top: 45px;
  width: 260px;
}

.victory-gif {
  box-shadow: 0 0 35px black, -3px 0 40px black,
    3px 0 40px black;
  height: 200px;
}

p {
  margin: 13px;
  padding-left: 35px;
}

.instruction {
  height: 50px;
  padding-top: 85px;
}

h2 {
  background-color: #6577c3e8;
  margin: 0;
  padding-top: 5px;
  text-align: center;
}

.guess-input {
  background-color: #6577c3e8;
  border-color: #6577c3e8;
  height: 32px;
  margin: 0;
  width: 210px;
}

.submit-guess {
  background-color: #6577c3e8;
  border-color: #6577c3e8;
  color: white;
  font-size: .8em;
  font-weight: bold;
  height: 37px;
  margin: 0;
  padding-top: 7px;
}

.wrong-guess {
  background-color: red;
}

.guess-input::placeholder {
  color: white;
}

.winning-message {
  background-color: black;
  box-shadow: 0 0 25px black, -3px 0 28px black,
    3px 0 28px black;
  font-size: 1.5em;
}