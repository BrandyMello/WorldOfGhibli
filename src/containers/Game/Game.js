import React, { Component } from 'react';
import './Game.scss';
import { connect } from 'react-redux';
import { setWins, setGuess } from '../../actions';
import { bindActionCreators } from 'redux';
import images from '../../images/images';
import PropTypes from 'prop-types';

export class Game extends Component { 
  constructor(props) {
    super(props);
    this.state = {
      guess: '',
      wins: this.props.wins,
      winningGif: '',
      isWinner: false,
      correct: true
    }
  }

  randomizeName = (name) => {
    var randomSort = () => {
      return Math.random() > .5 ? -1 : 1;
    }
    return name.split('').sort(randomSort).join('').toLowerCase();
  }

  handleGuess = (e) => {
    const guess = e.target.value;
    this.setState({ guess });
  }

  submitGuess = (e, name) => {
    e.preventDefault();
    this.props.setGuess(this.state.guess);
    this.checkGuess(this.state.guess, name);
  }

  checkGuess = (guess, name, index) => {
    if (guess.toLowerCase() === name.toLowerCase()) {
      this.setState({wins: this.state.wins += 1})
      this.props.setWins(this.state.wins);
      this.setState({winningGif: images[`film${index}`][2]});
      this.setState({isWinner: true});
      this.setState({ correct: true });
    } else {
      this.setState({correct: false});
    }
  }

  render() {
    const { index, name, gender, age, eyeColor, hairColor, wins } = this.props;
    const btnClass = this.state.correct ? 'submit-guess' : 'submit-guess wrong-guess'

  return (
    <main className="game-background-img whole-game">
      <article className="ea-game">
        {this.state.isWinner && 
        <>
          <h3 className="winning-message">Great Job!</h3>
          <img src={this.state.winningGif} alt='victory gif' className="victory-gif"></img>
        </> ||
        <div className='clue-div'>
          <h2>{this.randomizeName(name)}</h2>
          <p className="clues">CLUES:</p>
          <p>Gender: {gender}</p>
          <p>Age: {age}</p>
          <p>Eye Color: {eyeColor}</p>
          <p>Hair Color: {hairColor}</p>
        </div> }
        <form className="guess-form" onSubmit={(e) => this.submitGuess(e, name, index)}>
          <input className="guess-input" name="guess" placeholder="type guess here..." onChange={this.handleGuess}></input>
          <button className={btnClass}>GUESS</button>
          
        </form>
      </article>
    </main>
  )}
}

export const mapStateToProps = state => ({
  wins: state.wins,
  guess: state.guess
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({ setWins, setGuess }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Game);

Game.propTypes = {
  wins: PropTypes.number,
  guess: PropTypes.array,
  winningGif: PropTypes.string,
  isWinner: PropTypes.bool,
  correct: PropTypes.bool,
  setWins: PropTypes.func.isRequired,
  setGuess: PropTypes.func.isRequired
}