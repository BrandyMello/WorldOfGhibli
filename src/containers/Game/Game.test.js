import React from 'react';
import { Game, mapStateToProps, mapDispatchToProps } from './Game';
import { shallow } from 'enzyme';
import { setWins, setGuess } from '../../actions';

describe('Game', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Game
        index={3}
        key={234}
        name='Eboshi'
        gender='Male'
        age={14}
        eyeColor='brown'
        hairColor='black'
        characterNames={['char1', 'char2']}
        player='Brandy'
        setWins={jest.fn()}
        setGuess={jest.fn()}
      />
    )
  });

  it('should match the snapshot with the data passed through', () => {
    wrapper.randomizeName = jest.fn();
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('mapStateToProps should get the props', () => {
      const expectedProps = { wins: 0, guess: 'eboshi' }
      const mockState = { wins: 0, guess: 'eboshi' };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedProps)
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mappedDispatchedProps = mapDispatchToProps(mockDispatch);

    it('should call dispatch with the setWins action', () => {
      const dispatchSetWinsAction = setWins(0);

      mappedDispatchedProps.setWins(0);

      expect(mockDispatch).toHaveBeenCalledWith(dispatchSetWinsAction);
    });

    it('should call dispatch with the setGuess action', () => {
      const dispatchSetGuessAction = setGuess('eboshi');

      mappedDispatchedProps.setGuess('eboshi');

      expect(mockDispatch).toHaveBeenCalledWith(dispatchSetGuessAction);
    });
  });
});