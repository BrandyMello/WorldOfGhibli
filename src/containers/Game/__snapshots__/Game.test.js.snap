// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should match the snapshot with the data passed through 1`] = `
<main
  className="game-background-img whole-game"
>
  <article
    className="ea-game"
  >
    <div
      className="clue-div"
    >
      <h2>
        obiesh
      </h2>
      <p
        className="clues"
      >
        CLUES:
      </p>
      <p>
        Gender: 
        Male
      </p>
      <p>
        Age: 
        14
      </p>
      <p>
        Eye Color: 
        brown
      </p>
      <p>
        Hair Color: 
        black
      </p>
    </div>
    <form
      className="guess-form"
      onSubmit={[Function]}
    >
      <input
        className="guess-input"
        name="guess"
        onChange={[Function]}
        placeholder="type guess here..."
      />
      <button
        className="submit-guess"
      >
        GUESS
      </button>
    </form>
  </article>
</main>
`;
